core/src/main/scala/squid/quasi/ClassEmbedding.scala

```diff
@@ -182,7 +182,13 @@ class ClassEmbedding(override val c: whitebox.Context) extends QuasiMacros(c) {
     val clsDefTparams = clsDef.tparams map (stripVariance(_))
     
     // Makes the parameters acceptable for being lambda parameters: removes the `rhs`
-    def fixValDef(vd: ValDef) = ValDef(vd.mods, vd.name, vd.tpt, EmptyTree) // TODO handle by-name and repeated
+    def fixValDef(vd: ValDef) = vd.tpt match {
+      case AppliedTypeTree(Select(_, TypeName("<repeated>")), args) =>
+        val newTypeTree = tq"Seq[..$args]"
+        ValDef(vd.mods, vd.name, newTypeTree, EmptyTree)
+
+      case _ => ValDef(vd.mods, vd.name, vd.tpt, EmptyTree)
+    } // TODO handle by-name and repeated
```

Comment:
Can you move the comment back to the top of the `def` and make it `TODO handle by-name arguments`?